### PR TITLE
Updates Notification Center

### PR DESCRIPTION
**Pull request by NovacAndrei**

### CODE REVIEW

File: src/redux/wallets/actions/wallet-actions.ts [hunk 1]
```diff
@@ -866,3 +866,21 @@ export const setWalletsCredentials = (password: string) => async (
...
+    for (const wallet of Object.values(state.wallets)) {
+        for (const account of wallet.accounts) {
+            if (account.address?.toLocaleLowerCase() === address?.toLocaleLowerCase()) {
+                return {
+                    walletName: wallet.name,
+                    accountName: account.name || `Account ${account.index + 1}`
+                };
+            }
+        }
+    }
```
Comment (krisboit) [inline src/redux/wallets/actions/wallet-actions.ts:885]: you should return something if nothing is found.

File: src/core/utils/api-client/notifications.ts [hunk 2]
```diff
@@ -22,25 +23,37 @@ export class NotificationsApiClient {
...
+
+                const walletData: any = {
                     walletPublicKey,
                     timestamp: await getCurrentTimestampNTP(),
-                    domain: getWalletApiDomain(),
-                    page: page || 1
+                    domain: getWalletApiDomain()
                 };
 
                 const signature = getSignature(data, walletPrivateKey, walletPublicKey);
```
Comment (krisboit) [inline src/core/utils/api-client/notifications.ts:47]: you should use walletData for sign, right ? it's weird to sign all data object...

File: src/core/utils/api-client/notifications.ts [hunk 3]
```diff
@@ -22,25 +23,37 @@ export class NotificationsApiClient {
...
+                blockchainNetworks,
+                page: page || 1
+            };
+
+            for (const walletPublicKey of walletPublicKeys) {
+                const walletPrivateKey = await getWalletCredentialsKey(walletPublicKey);
+
+                const walletData: any = {
                     walletPublicKey,
                     timestamp: await getCurrentTimestampNTP(),
```
Comment (krisboit) [inline src/core/utils/api-client/notifications.ts:43]: does this have a cache ? or it does a ntp call every time ?

File: src/core/utils/api-client/notifications.ts [hunk 4]
```diff
@@ -175,24 +181,28 @@ export class NotificationsApiClient {
...
+                const walletData: any = {
                     walletPublicKey,
-                    signature
-                }
-            ];
+                    timestamp: await getCurrentTimestampNTP(),
+                    domain: getWalletApiDomain()
+                };
+
+                const signature = getSignature(data, walletPrivateKey, walletPublicKey);
```
Comment (krisboit) [inline src/core/utils/api-client/notifications.ts:201]: same here, only walletData should be signed

File: src/core/utils/api-client/notifications.ts (hunk 3 above)
Comment (NovacAndrei) [reply src/core/utils/api-client/notifications.ts:43]: It does make a ntp call every time.
Any suggestions on how to cache?

Comment (krisboit) [reply src/core/utils/api-client/notifications.ts:43]: yep, quite simple: `const timestamp = await getCurrentTimestampNTP()` before `for` :) this way you do it once :P

**Pull request merged**